src/server/scripts/Kalimdor/HallsOfOrigination/boss_isiset.cpp

```diff
@@ -0,0 +1,1071 @@
+                    DoStartMovement(me->GetVictim());
+                    break;
+                case EVENT_ASTRAL_RAIN:
+                    DoCast(me, SPELL_ASTRAL_RAIN_CONTROLLER);
+                    events.ScheduleEvent(EVENT_ASTRAL_RAIN, urand(15,19)*IN_MILLISECONDS);
+                    break;
+                case EVENT_ASTRAL_FAMILIAR:
+                    DoCast(me, SPELL_ASTRAL_FAMILIAR_CONTROLLER);
+                    events.ScheduleEvent(EVENT_ASTRAL_FAMILIAR, urand(26,30)*IN_MILLISECONDS);
+                    break;
+                case EVENT_VEIL_OF_SKY:
+                    DoCast(me, SPELL_MANA_SHIELD_CONTROLLER);
+                    events.ScheduleEvent(EVENT_VEIL_OF_SKY, urand(20,25)*IN_MILLISECONDS);
+                    break;
+                case EVENT_SUPERNOVA:
+                    DoCast(me, SPELL_SUPERNOVA);
+                    Talk(SAY_SUPERNOVA);
+                    Talk(SAY_SUPERNOVA_WARNING);
+                    events.ScheduleEvent(EVENT_SUPERNOVA, urand(45,50)*IN_MILLISECONDS);
+                    break;
+                case EVENT_IMAGES_ATTACK:
+                {
+                    DummyEntryCheckPredicate pred;
+                    summons.DoAction(ACTION_IMAGES_SET_AGGRESSIVE, pred);
+                    break;
+                }
+                case EVENT_DESPAWN_IMAGES:
+                {
+                    DummyEntryCheckPredicate pred;
+                    summons.DoAction(ACTION_IMAGES_DESPAWN, pred);
+                    instance->SetData(DATA_ISISET_PHASE, instance->GetData(DATA_ISISET_PHASE) + 1);
+                    me->RemoveAurasDueToSpell(SPELL_ASTRAL_SHIFT);
+                    RescheduleEvents();
+                    break;
+                }
+                default:
+                    break;
+                }
+            }
+
+            DoMeleeAttackIfReady();
+        }
+
+    private:
+        void RescheduleEvents()
+        {
+            events.Reset();
+            events.ScheduleEvent(EVENT_START_ATTACK, 1200);
+            if (instance->GetData(DATA_ISISET_ASTRAL_RAIN_ALIVE) == 1)
+                events.ScheduleEvent(EVENT_ASTRAL_RAIN, urand(4, 6)*IN_MILLISECONDS);
+            if (instance->GetData(DATA_ISISET_CELESTIAL_CALL_ALIVE) == 1)
+                events.ScheduleEvent(EVENT_ASTRAL_FAMILIAR, 7200);
+            if (instance->GetData(DATA_ISISET_VEIL_OF_SKY_ALIVE) == 1)
+                events.ScheduleEvent(EVENT_VEIL_OF_SKY, urand(8, 10)*IN_MILLISECONDS);
+            events.ScheduleEvent(EVENT_SUPERNOVA, urand(12, 17)*IN_MILLISECONDS);
+        }
+
+        // Mirror Image transition phase
+        void MirrorImage()
+        {
+            events.Reset();
+            instance->SendEncounterUnit(ENCOUNTER_FRAME_UPDATE_PRIORITY, me, 0);
+            mirrorImageDied = false;
+
+            DoCast(SPELL_ASTRAL_SHIFT);
+            DoCast(SPELL_MIRROR_IMAGE_STARRY_SKY_SPAWNER);
+            DoCast(SPELL_MIRROR_IMAGE_SPAWNER);
```

Comment:
has no category lock in this 3 spells? i mean, he can cast this 3 spells together without trigger flag?

Reply:
In sniffs, SPELL_MIRROR_IMAGE_SPAWNER is the last spell he casts, so it's not triggered, it triggers 3 other spells. Same goes for SPELL_MIRROR_IMAGE_STARRY_SKY_SPAWNER.